### PR TITLE
fix(menu-button): remove index for seperator item

### PR DESCRIPTION
**Pull request by saiponnada**
## Description
Added a separator count and deducted the previous separators count from the item's index and passed that value to reflect the correct index if list item. 

## References
#1792

### CODE REVIEW

File: src/components/ebay-menu/index.marko [hunk 1]
```diff
@@ -50,6 +51,9 @@ $ var baseClass = input.classPrefix || "menu";
                 var checked = component.isChecked(index);
             }
             <if (item.separator)>
+                $ {
+                    separatorCount += 1;
+                }
```
Comment (agliga) [inline src/components/ebay-menu/index.marko:56]: I think you can just do `$ separatorCount += 1;` in 1 line.

**Review by agliga — COMMENTED** [review]
Looks great. 
I would add a quick browser test with separators. Don't worry about updating the tests to using the new syntax for now.
Use the code to render separators in the server tests, render one with separators and then click an item before and after the separator, and verify the index is correct. 
Separator render: https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-menu/test/test.server.js#L37-L38
Copy this test https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-menu/test/test.browser.js#L79-L99 (you might have to check for the index as well)

**Review by ArtBlue — COMMENTED** [review]
There's something strange and inconsistent going on here.

1. The mouse-activated experience is not inline with the keyboard activated experience.
2. The keyboard activated experience seems to be skipping an item when using arrow keys.

https://user-images.githubusercontent.com/1675667/200351704-3d67593a-b162-4593-a3e1-e4778a4ee457.mp4

File: src/components/ebay-menu/index.marko [hunk 2]
```diff
@@ -50,6 +51,9 @@ $ var baseClass = input.classPrefix || "menu";
...
                 <hr class=`${baseClass}__separator` role="separator" />
```
Comment (ArtBlue) [inline src/components/ebay-menu/index.marko:57]: Is any `js` dependent on or interacting with this `<hr>`? If not, could we not just do a `tabindex="-1"` and an aria hidden on this and fix this without having to fiddle with indexes?

Comment (agliga) [reply src/components/ebay-menu/index.marko:57]: I think the issue came from treating these as actual item elements. We loop through all the items and if they are a separator, we render it as so.
My guess is the keyboard part might be triggering from makeup.

Comment (ianmcburnie) [reply src/components/ebay-menu/index.marko:57]: Seems to be working fine on makeup side (https://makeup.github.io/makeup-js/makeup-menu-button/index.html). Is this Safari?

I am currently re-working all of the makeup keyboard stuff and should hopefully be easier to debug when done.

Comment (ArtBlue) [reply src/components/ebay-menu/index.marko:57]: This was Chrome

**Comment by ArtBlue** [comment]
@saiponnada , also change the target from `master` to `15.2.0`.

File: src/components/ebay-menu/index.marko (hunk 2 above)
Comment (agliga) [reply src/components/ebay-menu/index.marko:57]: > First I used the mouse to click into the menu, and then I used the keyboard to navigate the items. Then I collapsed the menu and then used the keyboard to gain focus on the menu and hit <ENTER> and then navigated the menu items with arrows once more.

So i would do one thing
Check if this issue is present on current production
If it is, lets file an issue. If not then we need to fix it
At least the first scenario seems correct, when we activate without keyboard it wont show focus on chrome but then it does if we press down and the second item is selected
My guess is the second usecase is just showing where you left off but doesnt show the initial focus.

Comment (ArtBlue) [reply src/components/ebay-menu/index.marko:57]: Ok. Yea, it does seem like in the second scenario, it continues from where it left off, which is not ideal. It should really start from the beginning again after the menu is expanded again.

Comment (ArtBlue) [reply src/components/ebay-menu/index.marko:57]: Yep, it looks to be a production issue. I created the bug issue: https://github.com/eBay/ebayui-core/issues/1809

File: src/components/ebay-menu/index.marko (hunk 1 above)
Comment (saiponnada) [reply src/components/ebay-menu/index.marko:56]: Cool. Will do that.

**Comment by agliga** [comment]
@saiponnada So lets just change the separator iterator for now. Lets get this merged and then we can work on tests next week.

**Comment by saiponnada** [comment]
> @saiponnada So lets just change the separator iterator for now. Lets get this merged and then we can work on tests next week.
Agree @agliga . Will focus on code reviews for now.

File: src/components/ebay-menu/index.marko (hunk 1 above)
Comment (saiponnada) [reply src/components/ebay-menu/index.marko:56]: Forgot to push this change. Let me do this.

**Review by agliga — APPROVED** [review]

**Pull request merged**